src/object.rs

```diff
@@ -0,0 +1,334 @@
+//! Any APRS station can manually report the position of an APRS entity (e.g.
+//! another station or a weather phenomenon). This is intended for situations
+//! where the entity is not capable of reporting its own position.
+//! Object Reports specify an Object’s position, can have an optional timestamp,
+//! and can include course/speed information or other Extended Data. Object
+//! Reports are intended primarily for plotting the positions of moving objects
+//! (e.g. spacecraft, storms, marathon runners without trackers).
+//!
+//! Examples:
+//!
+
+use std::convert::TryFrom;
+use std::io::Write;
+
+use Callsign;
+use DecodeError;
+
+use EncodeError;
+use Timestamp;
+
+use crate::{AprsCompressedCs, AprsCst, Latitude, Longitude, Precision};
+
+#[derive(Clone, Debug, PartialEq)]
+pub struct AprsObject {
+    pub to: Callsign,
+    pub name: String,
+    pub live: bool,
+    pub timestamp: Timestamp,
+    pub latitude: Latitude,
+    pub longitude: Longitude,
+    pub precision: Precision,
+    pub symbol_table: char,
+    pub symbol_code: char,
+    pub comment: Vec<u8>,
+    pub cst: AprsCst,
+}
+
+impl AprsObject {
+    pub fn decode(b: &[u8], to: Callsign) -> Result<Self, DecodeError> {
+        // format for uncompressed is
+        // N Bytes      Description/Value
+        // -----------------------------
+        // [9]          (object name, space-padded)
+        // [1]          (object live =  '*' dead = '_')
+        // [7]          (timestamp)
+        // [8]          (latitude)
+        // [1]          (sym table id)
+        // [9]          (longitude)
+        // [1]          (sym code)
+        // [7/0]        (extended data [NOT PARSING])
+        // [0-36/0-43]  (comment)
+        //
+
+        // format for compressed is
+        // N Bytes      Description/Value
+        // -----------------------------
+        // [9]          (object name, space-padded)
+        // [1]          (object live =  '*' dead = '_')
+        // [7]          (timestamp)
+        // [13]         (compressed position data)
+        // [43]         (comment)
+
+        // packet must have at least name + liveness + timestamp --> (9+1+7) so check that here
+
+        if b.len() < 17 {
+            return Err(DecodeError::InvalidObjectFormat(
+                b.to_vec(),
+                format!(
+                    "packet length must be at least 17 bytes long, got {}",
+                    b.len()
+                ),
+            ));
+        }
+
+        let name = b
+            .get(..9)
+            .map(|by| String::from_utf8_lossy(by).trim().to_string())
+            .unwrap();
```

Comment:
Not sure about this. I think we probably want to keep this as a `Vec<u8>` like we do for APRS messages, since it's possible the name may not be valid UTF-8

Reply:
There's a `trim_spaces_end` method in `message.rs` that we could move and use here